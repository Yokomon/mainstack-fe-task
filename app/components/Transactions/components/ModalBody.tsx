"use client";

import React, { useState } from "react";
import { DateRangePicker } from "react-nice-dates";
import { enGB } from "date-fns/locale";

import SelectDropdown from "../../MultiSelectDropdown";
import {
  getTransactionStatusOptions,
  getTransactionTypeOptions,
} from "@/app/utils";
import ModalButton from "./ModalButton";

const ModalBody = () => {
  const [startDate, setStartDate] = useState();
  const [endDate, setEndDate] = useState();

  const trasactionOptions = getTransactionTypeOptions();
  const transactionStatusOptions = getTransactionStatusOptions();

  return (
    <div className="space-y-5 py-2" data-testid="modalbody">
      <div className="flex items-center justify-between">
        <ModalButton text="Today" />
        <ModalButton text="Last 7 days" />
        <ModalButton text="This month" />
        <ModalButton text="Last 3 months" />
      </div>
      <div className="my-5 space-y-2">
        <h3>Date Range</h3>
        <div className="relative flex items-center justify-between">
          <DateRangePicker
            startDate={startDate}
            endDate={endDate}
            //@ts-expect-error
            onStartDateChange={setStartDate}
            //@ts-expect-error
            onEndDateChange={setEndDate}
            minimumDate={new Date()}
            minimumLength={1}
            format="dd MMM yyyy"
            locale={enGB}
          >
            {({ startDateInputProps, endDateInputProps, focus }) => (
              <div className="space-x-8">
                <input {...startDateInputProps} placeholder="Start date" />
                <input {...endDateInputProps} placeholder="End date" />
              </div>
            )}
          </DateRangePicker>
        </div>
      </div>
      <SelectDropdown options={trasactionOptions} name="Transaction Type" />
      <SelectDropdown
        options={transactionStatusOptions}
        name="Transaction Status"
      />
    </div>
  );
};

export default ModalBody;
