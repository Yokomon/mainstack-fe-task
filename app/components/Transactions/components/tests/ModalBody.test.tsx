import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import ModalBody from "../ModalBody";

jest.mock("react-nice-dates", () => ({
  DateRangePicker: ({ children }: any) =>
    children({
      startDateInputProps: { value: "", onChange: () => {} },
      endDateInputProps: { value: "", onChange: () => {} },
      focus: null,
    }),
}));

jest.mock("../../../MultiSelectDropdown", () => ({
  __esModule: true,
  default: () => <div data-testid="mockSelectDropdown" />,
}));

jest.mock("../ModalButton", () => ({
  __esModule: true,
  default: ({ text }: any) => (
    <button data-testid={`modalButton-${text}`} onClick={() => {}}>
      {text}
    </button>
  ),
}));

describe("ModalBody Component", () => {
  it("renders without crashing", () => {
    render(<ModalBody />);
    expect(screen.getByTestId("modalbody")).toBeInTheDocument();
  });

  it("displays date range inputs", () => {
    render(<ModalBody />);
    expect(screen.getByPlaceholderText("Start date")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("End date")).toBeInTheDocument();
  });

  it("displays modal buttons", () => {
    render(<ModalBody />);
    expect(screen.getByTestId("modalButton-Today")).toBeInTheDocument();
    expect(screen.getByTestId("modalButton-Last 7 days")).toBeInTheDocument();
    expect(screen.getByTestId("modalButton-This month")).toBeInTheDocument();
    expect(screen.getByTestId("modalButton-Last 3 months")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<ModalBody />);
    expect(asFragment()).toMatchSnapshot();
  });
});
