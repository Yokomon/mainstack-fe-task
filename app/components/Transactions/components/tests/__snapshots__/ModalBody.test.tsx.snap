// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBody Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="space-y-5 py-2"
    data-testid="modalbody"
  >
    <div
      class="flex items-center justify-between"
    >
      <button
        data-testid="modalButton-Today"
      >
        Today
      </button>
      <button
        data-testid="modalButton-Last 7 days"
      >
        Last 7 days
      </button>
      <button
        data-testid="modalButton-This month"
      >
        This month
      </button>
      <button
        data-testid="modalButton-Last 3 months"
      >
        Last 3 months
      </button>
    </div>
    <div
      class="my-5 space-y-2"
    >
      <h3>
        Date Range
      </h3>
      <div
        class="relative flex items-center justify-between"
      >
        <div
          class="space-x-8"
        >
          <input
            placeholder="Start date"
            value=""
          />
          <input
            placeholder="End date"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      data-testid="mockSelectDropdown"
    />
    <div
      data-testid="mockSelectDropdown"
    />
  </div>
</DocumentFragment>
`;
